cfg/config.go

```diff
@@ -266,18 +266,24 @@ func Parse(ptr interface{}, values map[string]int) error {
 // 	}
 // }
 // this function will set `config.Child.T = config.T`
-// see file.d/cfg/config_test.go:TestHierarchy for an example
+// see file.d/cfg/config_test.go:TestHierarchy for an example.
 func ParseChild(parent reflect.Value, v reflect.Value, values map[string]int) error {
 	if v.CanAddr() {
-		for i := 0; i < v.NumField(); i++ {
-			name := v.Type().Field(i).Name
+		// Check for cfg pointers.
```

Comment:
Not retated to your PR, but every time I look at this parsing it makes me sad. 
Why do we do black magic with something so simple as parsing a configuration  :(

Reply:
Custom tags `child`, `slice` requires this